src/rpdk/core/contract/resource_client.py

```diff
@@ -267,9 +267,39 @@ def generate_invalid_create_example(self):
         example = self.invalid_strategy.example()
         return override_properties(example, self._overrides.get("CREATE", {}))
 
+    def get_unique_keys_for_model(self, create_model):
+        return {
+            k: v
+            for k, v in create_model.items()
+            if self.is_property_in_path(k, self.primary_identifier_paths)
```

Comment:
does this work for nested properties? say we have a schema 
```
{
    "properties": {
        "a": {"type": "number"},
        "b": {"type": "number"},
        "c": {"type": "object",
                "properties": {
                     "e": {"type": "string"}
                 }
               },
        "d": {"type": "number"},
    },
    "readOnlyProperties": ["/properties/b"],
    "createOnlyProperties": ["/properties/c"],
    "primaryIdentifier": ["/properties/c/e"],
    "writeOnlyProperties": ["/properties/d"],
}
```

Will we be able to copy over the c/e property? it seems like we only do a check on the resource model's top level keys. It seems like it would currently include the entire c object and not just the c/e key

Reply:
That is a good catch and indeed, it will not work correctly in that case. In my opinion, this is an issue due to the construction of primary_identifier_paths rather than the code this PR is addressing. It is something we should log as a bug to fix, possibly across all  <value>_identifier_paths we are creating.